### ISSUE TITLE
Issue: Running ACT on BigSur 11.2.2 fails due to Mounts denied on /github/workspace

### ISSUE THREAD

**Issue by ian-lewis-d** [nektos/act#551]
## Act version

act version 0.2.20

## Expected behaviour

Running $ act -j vetting should run the vetting section in .github/workflows/build.xml

## Actual behaviour

The process runs stops with the following message:

```
[Build/Vetting        ] Extracting content to '/github/'
[Build/Vetting        ] Exec command '[bash --noprofile --norc -eo pipefail /github/workflow/1]'
| docker: Error response from daemon: Mounts denied: 
| The path /github/workspace
| is not shared from OS X and is not known to Docker.
| You can configure shared paths from Docker -> Preferences... -> File Sharing.
| See https://docs.docker.com/docker-for-mac/osxfs/#namespaces for more info.
| .
| ERRO[0000] error waiting for container: context canceled 
```

This is on a Mac OS BigSur 11.2.2 with the default Sandboxing enabled.

I have attempted downgrading Docker Desktop (to 2.5.0.1 from 3.1 latest), Turning off gRPC Fuse and none of this gets me anywhere.

It seems, from what I can work out, that the BigSur MacOS has made root a Read Only file system so adding /github/workflow will fail. This leads to the mount failing as there is no mount point. This is my current theory but I have only limited experience with Docker, Github Actions and Mac computers.

## Workflow and/or repository

<details>
  <summary>workflow</summary>

```none
name: example workflow

on: [push]

jobs:
  vetting:
    name: Vetting
    runs-on: ubuntu-latest
    timeout-minutes: 5
    steps:
      - name: Checkout Repo
        uses: actions/checkout@v2
      - name: Project Validation
        run: docker run --rm -v $(pwd):/app ******/devtools:project-validator
```

</details>

## Steps to reproduce

1. Clone repo (private)
2. Enter cloned repo directory
3. Run `act -j vetting`

## `act` output

<details>
  <summary>Log</summary>

```
DEBU[0000] Loading environment from /Users/user/projects/[REDACTED]/.env 
DEBU[0000] Loading secrets from /Users/user/projects/[REDACTED]/.secrets 
DEBU[0000] Loading workflows from '/Users/user/projects/[REDACTED]/.github/workflows' 
DEBU[0000] Reading workflow '/Users/user/projects/[REDACTED]/.github/workflows/build.yml' 
DEBU[0000] Planning event: push                         
DEBU[0000] Loading slug from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] Loading revision from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] HEAD points to '[REDACTED]' 
DEBU[0000] HEAD matches refs/heads/master               
DEBU[0000] using github ref: refs/heads/master          
DEBU[0000] context env => map[ACT:true]                 
DEBU[0000] Loading slug from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] Loading revision from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] HEAD points to '[REDACTED]' 
DEBU[0000] HEAD matches refs/heads/master               
DEBU[0000] using github ref: refs/heads/master          
DEBU[0000] context env => map[ACT:true GITHUB_TOKEN:${{ secrets.ORG_GITHUB_TOKEN }} GOPRIVATE:github.com/[REDACTED]] 
DEBU[0000] Loading slug from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] Loading revision from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] HEAD points to '[REDACTED]' 
DEBU[0000] HEAD matches refs/heads/master               
DEBU[0000] using github ref: refs/heads/master          
DEBU[0000] context env => map[ACT:true GITHUB_TOKEN:${{ secrets.ORG_GITHUB_TOKEN }}] 
DEBU[0000] Loading slug from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] Loading revision from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] HEAD points to '[REDACTED]' 
DEBU[0000] HEAD matches refs/heads/master               
DEBU[0000] using github ref: refs/heads/master          
DEBU[0000] context env => map[ACT:true ECR_REGISTRY:[REDACTED].dkr.ecr.eu-west-1.amazonaws.com GITHUB_TOKEN:${{ secrets.ORG_GITHUB_TOKEN }} ORG_AWS_ACCESS_KEY_ID:${{ secrets.ORG_AWS_ACCESS_KEY_ID }} ORG_AWS_SECRET_ACCESS_KEY:${{ secrets.ORG_AWS_SECRET_ACCESS_KEY }}] 
[Build/Vetting        ] 🚀  Start image=catthehacker/ubuntu:act-latest
DEBU[0000] Loading slug from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] Loading revision from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] HEAD points to '[REDACTED]' 
DEBU[0000] HEAD matches refs/heads/master               
DEBU[0000] using github ref: refs/heads/master          
[Build/Vetting        ]   🐳  docker pull catthehacker/ubuntu:act-latest
DEBU[0000] Image exists? true                           
[Build/Vetting        ] Removed container: e8124f9b1a15e85d24f96165439b615fc846ed3b8ddf17cac24d83cb4c081fbf
[Build/Vetting        ]   🐳  docker volume rm act-Build-Vetting
[Build/Vetting        ]   🐳  docker create image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Build/Vetting        ] Created container name=act-Build-Vetting id=80688ddcc02db7cfb1ea28ad94ed41c11ff8adb5c71198c38f3da1b767447eeb from image catthehacker/ubuntu:act-latest
[Build/Vetting        ] ENV ==> [RUNNER_TOOL_CACHE=/opt/hostedtoolcache RUNNER_OS=Linux RUNNER_TEMP=/tmp]
[Build/Vetting        ]   🐳  docker run image=catthehacker/ubuntu:act-latest entrypoint=["/usr/bin/tail" "-f" "/dev/null"] cmd=[]
[Build/Vetting        ] Starting container: 80688ddcc02db7cfb1ea28ad94ed41c11ff8adb5c71198c38f3da1b767447eeb
[Build/Vetting        ] Started container: 80688ddcc02db7cfb1ea28ad94ed41c11ff8adb5c71198c38f3da1b767447eeb
[Build/Vetting        ]   🐳  docker cp src=/Users/user/projects/[REDACTED]/. dst=/github/workspace
[Build/Vetting        ] Exec command '[mkdir -p /github/workspace]'
DEBU[0000] Writing tarball /var/folders/2v/xshx_qk10j1gtx2gb928kszw0000gn/T/act370122493 from /Users/user/projects/[REDACTED]/. 
DEBU[0000] Stripping prefix:/Users/user/projects/[REDACTED]/ src:/Users/user/projects/[REDACTED]. 
[Build/Vetting        ] Extracting content from '/var/folders/2v/xshx_qk10j1gtx2gb928kszw0000gn/T/act370122493' to '/github/workspace'
DEBU[0000] Writing entry to tarball workflow/event.json len:2 
DEBU[0000] Writing entry to tarball workflow/envs.txt len:0 
DEBU[0000] Writing entry to tarball home/.act len:0     
[Build/Vetting        ] Extracting content to '/github/'
DEBU[0000] Loading slug from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] Loading revision from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0000] Found revision: [REDACTED] 
DEBU[0000] HEAD points to '[REDACTED]' 
DEBU[0000] HEAD matches refs/heads/master               
DEBU[0000] using github ref: refs/heads/master          
DEBU[0001] Loading slug from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0001] Found revision: [REDACTED] 
DEBU[0001] Loading revision from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0001] Found revision: [REDACTED] 
DEBU[0001] HEAD points to '[REDACTED]' 
DEBU[0001] HEAD matches refs/heads/master               
DEBU[0001] using github ref: refs/heads/master          
DEBU[0001] context env => map[ACT:true]                 
DEBU[0001] context env => map[ACT:true CI:true GITHUB_ACTION:0 GITHUB_ACTIONS:true GITHUB_ACTOR:nektos/act GITHUB_API_URL:https://api.github.com GITHUB_ENV:/github/workflow/envs.txt GITHUB_EVENT_NAME:push GITHUB_EVENT_PATH:/github/workflow/event.json GITHUB_GRAPHQL_URL:https://api.github.com/graphql GITHUB_REF:refs/heads/master GITHUB_REPOSITORY:[REDACTED]/[REDACTED] GITHUB_RUN_ID:1 GITHUB_RUN_NUMBER:1 GITHUB_SERVER_URL:https://github.com GITHUB_SHA:[REDACTED]
 GITHUB_TOKEN: GITHUB_WORKFLOW:Build GITHUB_WORKSPACE:/github/workspace HOME:/github/home] 
DEBU[0001] setupEnv: map[ACT:true CI:true GITHUB_ACTION:0 GITHUB_ACTIONS:true GITHUB_ACTOR:nektos/act GITHUB_API_URL:https://api.github.com GITHUB_ENV:/github/workflow/envs.txt GITHUB_EVENT_NAME:push GITHUB_EVENT_PATH:/github/workflow/event.json GITHUB_GRAPHQL_URL:https://api.github.com/graphql GITHUB_REF:refs/heads/master GITHUB_REPOSITORY:[REDACTED] GITHUB_RUN_ID:1 GITHUB_RUN_NUMBER:1 GITHUB_SERVER_URL:https://github.com GITHUB_SHA:[REDACTED] GITHUB_TOKEN: GITHUB_WORKFLOW:Build GITHUB_WORKSPACE:/github/workspace HOME:/github/home] 
[Build/Vetting        ] ⭐  Run Checkout Repo
DEBU[0001] Loading slug from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0001] Found revision: [REDACTED] 
DEBU[0001] Loading revision from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0001] Found revision: [REDACTED] 
DEBU[0001] HEAD points to '[REDACTED]' 
DEBU[0001] HEAD matches refs/heads/master               
DEBU[0001] using github ref: refs/heads/master          
[Build/Vetting        ] Skipping actions/checkout
[Build/Vetting        ]   ✅  Success - Checkout Repo
DEBU[0001] Loading slug from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0001] Found revision: [REDACTED] 
DEBU[0001] Loading revision from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0001] Found revision: [REDACTED] 
DEBU[0001] HEAD points to '[REDACTED]' 
DEBU[0001] HEAD matches refs/heads/master               
DEBU[0001] using github ref: refs/heads/master          
DEBU[0001] Loading slug from git directory '/Users/user/projects/[REDACTED].git' 
DEBU[0001] Found revision: [REDACTED] 
DEBU[0001] Loading revision from git directory '/Users/user/projects/[REDACTED]/.git' 
DEBU[0001] Found revision: [REDACTED] 
DEBU[0001] HEAD points to '[REDACTED]' 
DEBU[0001] HEAD matches refs/heads/master               
DEBU[0001] using github ref: refs/heads/master          
DEBU[0001] context env => map[ACT:true]                 
DEBU[0001] context env => map[ACT:true CI:true GITHUB_ACTION:1 GITHUB_ACTIONS:true GITHUB_ACTOR:nektos/act GITHUB_API_URL:https://api.github.com GITHUB_ENV:/github/workflow/envs.txt GITHUB_EVENT_NAME:push GITHUB_EVENT_PATH:/github/workflow/event.json GITHUB_GRAPHQL_URL:https://api.github.com/graphql GITHUB_REF:refs/heads/master GITHUB_REPOSITORY:[REDACTED] GITHUB_RUN_ID:1 GITHUB_RUN_NUMBER:1 GITHUB_SERVER_URL:https://github.com GITHUB_SHA:[REDACTED] GITHUB_TOKEN: GITHUB_WORKFLOW:Build GITHUB_WORKSPACE:/github/workspace HOME:/github/home] 
DEBU[0001] setupEnv: map[ACT:true CI:true GITHUB_ACTION:1 GITHUB_ACTIONS:true GITHUB_ACTOR:nektos/act GITHUB_API_URL:https://api.github.com GITHUB_ENV:/github/workflow/envs.txt GITHUB_EVENT_NAME:push GITHUB_EVENT_PATH:/github/workflow/event.json GITHUB_GRAPHQL_URL:https://api.github.com/graphql GITHUB_REF:refs/heads/master GITHUB_REPOSITORY:[REDACTED]/[REDACTED] GITHUB_RUN_ID:1 GITHUB_RUN_NUMBER:1 GITHUB_SERVER_URL:https://github.com GITHUB_SHA:[REDACTED] GITHUB_TOKEN: GITHUB_WORKFLOW:Build GITHUB_WORKSPACE:/github/workspace HOME:/github/home] 
[Build/Vetting        ] ⭐  Run Project Validation
DEBU[0001] Wrote command 'docker run --rm -v $(pwd):/app [REDACTED]/devtools:project-validator' to 'workflow/1' 
DEBU[0001] Writing entry to tarball workflow/1 len:64   
[Build/Vetting        ] Extracting content to '/github/'
[Build/Vetting        ] Exec command '[bash --noprofile --norc -eo pipefail /github/workflow/1]'
| docker: Error response from daemon: Mounts denied: 
| The path /github/workspace
| is not shared from OS X and is not known to Docker.
| You can configure shared paths from Docker -> Preferences... -> File Sharing.
| See https://docs.docker.com/docker-for-mac/osxfs/#namespaces for more info.
| .
| ERRO[0000] error waiting for container: context canceled 
[Build/Vetting        ]   ❌  Failure - Project Validation
DEBU[0001] exit with `FAILURE`: 125                     
DEBU[0001] exit with `FAILURE`: 125                     
DEBU[0001] exit with `FAILURE`: 125                     
DEBU[0001] exit with `FAILURE`: 125                     
Error: exit with `FAILURE`: 125

```

</details>

**Comment by catthehacker** [comment]
This looks like a duplicate of #410.

**Comment by ian-lewis-d** [comment]
@catthehacker I tried that workaround and it did not work for me.

As far as I can tell the /github/workspace mkdir command is hardcoded which means it will always fail if docker cannot access the read-only root on BugSur.

**Comment by catthehacker** [comment]
The problem is not with Mac, mounting directories to docker container running in docker container is currently not supported.

**Comment by ian-lewis-d** [comment]
Thank you. Is that spelled out anywhere in the documentation?

**Comment by ian-lewis-d** [comment]
I see there's a `privileged` mode which allows docker to run inside docker. Would it be possible to apply this to act when it runs?

**Comment by catthehacker** [comment]
You can but it will fail, because the docker command you are executing in workflow doesn't know that it's already in docker container.
It will try to map path from `act` docker container to your container but it should map path from host to your container.
[Sam already explained it perfectly](https://github.com/nektos/act/issues/410#issuecomment-732096442)

**Comment by catthehacker** [comment]
This should be resolved already by #567
